### ISSUE TITLE
Arabic characters are not announced with input help

### ISSUE THREAD

**Issue by mohdshara** [nvaccess/nvda#7317]
steps to reproduce:
1. switch to the Arabic keyboard layout.
2. activate Input help by pressing NVDA+1
3. press character keys.
NVDA reads the English key. For Example, typing the Arabic letter ش NVDA will announce the English letter a. 
an exception to this are those keys that do not produce English characters. Example [ ] ; ' l / . those will announced the Arabic character that they type.

Nvda 2017.2 Windows 10 Creator's update.

**Comment by jcsteh** [comment]
Do you rely on English characters for NVDA commands; e.g. h for next heading? The problem with changing this is that NVDA commands which are based on English keys will stop working. While it seems strange to rely on English keys for a non-English keyboard layout, it seems that NVDA is not alone here and translators in the past have expressed a desire to keep things like this. For example, on some keyboard layouts, the number 1 is shifted, but this would mean that Input Help becomes NVDA+shift+1 and there'd be no way to jump to previous level 1 heading. It'd also mean translators would have to rebind many keys for their languages.

**Comment by mohdshara** [comment]
Yes we do. and you're right about that. however the use case for those users is the ability to learn the keys. my life, for one, will become very difficult if I have to learn another set of keys for the Arabic layout, given that their translation commonly doesn't make a lot of sense for someone who have always dealt with those in their English terms. what is the situation like for users of other languages? do they hear the English key?

**Comment by ruifontes** [comment]
In portuguese, Portugal and Brasil, we use the english characters.

Rui


-----Mensagem Original----- 
De: Mohammed Al Shara
Data: 26 de junho de 2017 09:08
Para: nvaccess/nvda
Cc: Subscribed
Assunto: Re: [nvaccess/nvda] Arabic characters are not announced with input 
help (#7317)



Yes we do. and you're right about that. however the use case for those users 
is the ability to learn the keys. my life, for one, will become very 
difficult if I have to learn another set of keys for the Arabic layout, 
given that their translation commonly doesn't make a lot of sense for 
someone who have always dealt with those in their English terms. what is the 
situation like for users of other languages? do they hear the English key?

**Comment by bhavyashah** [comment]
Since Input Help is already context-sensitive, i.e. it does not report single letter navigation if I was in Input Help mode while focused on the Desktop, I think there is a way to clearly define this bug in certain contexts. If I have switched to a non-English keyboard layout (Hindi (India) - Hindi Traditional in my case), then enter into focus mode in an edit field, then activate Input Help mode, and then press the alphabet keys, I think it is reasonable to expect the non-English (Hindi in my case) characters that would have gotten inputted in the edit box to be spoken as opposed to the English characters. The actual result is that NVDA speaks the English characters in Input Help. Thoughts appreciated.
